File: crates/nil/src/command/cheat/city.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use crate::error::Result;
use crate::manager::ManagerExt;
use nil_core::city::Stability;
use nil_core::continent::Coord;
use tauri::AppHandle;

#[tauri::command]
pub async fn cheat_set_stability(app: AppHandle, coord: Coord, stability: Stability) -> Result<()> {
  app
    .client(async |cl| {
      cl.cheat_set_stability(coord, stability)
        .await
    })
    .await?
    .map_err(Into::into)
}

File: app/src/types/core/city.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { Ruler } from '@/types/core/ruler';
import type { Coord } from '@/types/core/continent';

export interface PublicCity {
  readonly coord: Coord;
  readonly name: string;
  readonly owner: Ruler;
}

export interface City extends PublicCity {
  readonly infrastructure: Infrastructure;
  readonly stability: number;
}
